django/contrib/postgres/fields/array.py

```diff
@@ -157,6 +157,12 @@ def get_transform(self, name):
         else:
             return SliceTransformFactory(start, end)
 
+    def slice_expression(self, expression, start, length):
+        # If length is not provided, don't specify an end to slice to the end
+        # of the array.
+        end = '' if length is None else start + length - 1
```

Comment:
Is there a reason (there might well be!) why end can be either an empty string (`''`) _or_ a number (I'm assuming)? I assume the empty string is a falsy sentinel at first glance, but I may be mis-reading. It _seems_ like an awkward condition to potentially require user-subclasses to support, but if it's a technical necessity, so be it.

Reply:
Yes, because we can then get slicing in PostgreSQL such as `[5:]` because the `end` value being `''` allows nothing to be substituted.

Arguably it isn't nice to mix strings and integers. We could use `None` instead, but then we'd need to explicitly handle that case later.

Reply:
Ah I thought everywhere `self.end` was used was already guarded by a falsy check, which `0` or `None` would also satisfy, my bad.

Reply:
Sorry @felixxm. I was making my comments on this from memory and hadn't checked.

I wonder if this could now indeed be `None` instead of `''` as looking at the implementation of `SliceTransform.as_sql()`, it already has a separate `if self.end:` branch being added in this patch.

(If that branch hadn't been added, then the original unmodified version there would have worked with the `''` value. But perhaps that is too magic and `int | None` is nicer than `int | str`.)